eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -163,6 +163,13 @@ jobs:
 
       - template: ../steps/upload-repository-on-failure.yml
 
+      - ${{ each artifact in parameters.Artifacts }}:
+        - task: PublishPipelineArtifact@1
+          inputs:
+            targetPath: '$(Pipeline.Workspace)/sdk/${{ parameters.ServiceDirectory }}/${{ artifact.name }}/target/oom.hprof'
+            artifactType: 'filepath'
```

Comment:
This seems wrong. Artifacts is the list of things built that should be published, not a list of destinations to publish to.
If the goal is to publish the artifact as a pipeline artifact, retained in Azure DevOps, you should replace `inputs` with something like:
```yaml
  condition: 
  inputs:
    targetPath: $(Pipeline.Workspace)/sdk/${{ parameters.ServiceDirectory }}/${{ artifact.name }}/target/oom.hprof
    artifactType: pipeline   #this line isn't necessary as pipeline is the default
    artifactName: '${{ artifact.name }}.oom.hprof'
```

If the oom.hprof is a memory dump and we have reason to believe it may contain sensitive information, e.g. credentials, we should consider more secure destinations for the files, like blob storage.

`filepath` base artifact publishing appears to be targeting DevOps users with private, network connected agents with fileshare access, i.e.   \\someserver\share\